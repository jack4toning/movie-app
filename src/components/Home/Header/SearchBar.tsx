import React from 'react';
import styled from 'styled-components';

export default function SearchBar() {
  return (
    <Container>
      <Title>FIND YOUR MOVIE</Title>
      <Input placeholder='What do you want to watch?' autoFocus />
      <Button>SEARCH</Button>
    </Container>
  );
}

const Container = styled.div`
  position: absolute;
  top: 101px;
  left: 120px;
`;

const Title = styled.p`
  font-weight: 300;
  font-size: 40px;
  margin: 0 0 38px 0;
`;

const Input = styled.input`
  width: 713px;
  height: 57px;
  background: rgba(50, 50, 50, 0.8);
  opacity: 0.7;
  border-radius: 4px;
  border: none;
  outline: none;
  text-indent: 1rem;
  caret-color: #d4d4d4;
  font-size: 20px;

  ::-webkit-input-placeholder {
    /*Webkit browsers*/
    font-family: 'Montserrat', sans-serif;
  }
`;

const Button = styled.button`
  width: 233px;
  height: 57px;
  margin: 0 0 0 14px;
  background-color: #f65261;
  border-radius: 4px;
  border: none;
  font-size: 20px;
  font-weight: 500;
  color: #fff;
  cursor: pointer;

  :hover {
    background-color: #eb4553;
  }
`;
