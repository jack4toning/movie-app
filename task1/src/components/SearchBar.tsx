import React from 'react';
import styled from 'styled-components';

export class SearchBar extends React.Component<any, { keyword: string }> {
  constructor(props: any) {
    super(props);
    this.state = { keyword: '' };
  }

  handleInput(keyword: string) {
    this.setState({ keyword });
  }

  render() {
    return (
      <Container>
        <CustomInput
          placeholder='What do you want to watch?'
          autoFocus
          onChange={e => {
            this.handleInput(e.target.value);
          }}
        />
        <CustomButton>SEARCH</CustomButton>
      </Container>
    );
  }
}

const Container = styled.div`
  width: 1080px;
  height: 600px;
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: #232323;
  border-radius: 4px;
  margin-bottom: 24px;
`;

const CustomInput = styled.input`
  width: 713px;
  height: 57px;
  background: rgba(50, 50, 50, 0.8);
  opacity: 0.7;
  border-radius: 4px;
  border: none;
  outline: none;
  text-indent: 1rem;
  color: #fff;
  caret-color: #d4d4d4;
  font-size: 20px;
  font-weight: 400;
`;

const CustomButton = styled.button`
  width: 233px;
  height: 57px;
  margin: 0 0 0 14px;
  background-color: #f65261;
  border-radius: 4px;
  border: none;
  font-size: 20px;
  font-weight: 500;
  color: #ffffff;
`;
