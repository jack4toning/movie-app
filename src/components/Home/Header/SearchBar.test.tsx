import { fireEvent, screen, render } from '@testing-library/react';
import SearchBar from './SearchBar';
import { BrowserRouter as Router } from 'react-router-dom';

describe('SearchBar', () => {
  it('should render same results each time no matter what other code changes', () => {
    const { container } = render(
      <Router>
        <SearchBar />
      </Router>
    );
    expect(container).toMatchSnapshot();
  });

  test('url should change to /search/"searchValue" when click search button', () => {
    render(
      <Router>
        <SearchBar />
      </Router>
    );
    fireEvent.change(
      screen.getByPlaceholderText('What do you want to watch?'),
      {
        target: { value: 'Shawshank' },
      }
    );
    fireEvent.click(screen.getByRole('button'));
    console.log(window.location.pathname);
    expect(window.location.pathname).toBe('/search/Shawshank');
  });
});
