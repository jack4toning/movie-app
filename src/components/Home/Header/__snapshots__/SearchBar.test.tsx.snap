// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar should render same results each time no matter what other code changes 1`] = `
<div>
  <div
    class="sc-bcXHqe eFjOgP"
  >
    <p
      class="sc-gswNZR fseLN"
    >
      FIND YOUR MOVIE
    </p>
    <div
      class="sc-dkrFOg hSvpyl"
    >
      <input
        class="sc-eDvSVe hajJtn"
        placeholder="What do you want to watch?"
        value=""
      />
    </div>
    <button
      class="sc-jSUZER fpPOHU"
    >
      SEARCH
    </button>
  </div>
</div>
`;
